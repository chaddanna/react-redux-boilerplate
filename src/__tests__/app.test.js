import { mount, shallow } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';

import Component from '../app';

describe('Component', () => {
  let props;

  describe('Smoke tests', () => {
    it('snapshot test', () => {
      expect(
        renderer.create(<Component { ...props } />).toJSON()
      ).toMatchSnapshot();
    });

    it('renders without crashing', () => {
      shallow(<Component { ...props } />);
    });
  });

  describe('Mounted tests', () => {
    let mountedComponent;
    const getComponent = () => {
      if (!mountedComponent) {
        mountedComponent = mount(<Component { ...props } />).find(Component);
      }

      return mountedComponent;
    };

    beforeEach(() => {
      mountedComponent = undefined;
    });

    it("always renders 'Main'", () => {
      expect(getComponent().find('Main')).toHaveLength(1);
    });
  });
});
