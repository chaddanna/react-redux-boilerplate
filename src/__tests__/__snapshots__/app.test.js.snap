// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Smoke tests snapshot test 1`] = `
<div
  className="main-view"
>
  <header
    className="header"
  >
    <img
      alt="logo"
      className="logo"
      src="logo.svg"
    />
    <h1
      className="title"
    >
      Main View
    </h1>
  </header>
  <div
    className="content-block"
  >
    <p>
      Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
    </p>
  </div>
</div>
`;
